src/coprocessor/dag/expr/mod.rs

```diff
@@ -264,10 +286,28 @@ impl Expression {
                     .map(|child| Expression::build(ctx, child))
                     .collect::<Result<Vec<_>>>()
                     .map(|children| {
+                        let init_rng = match expr.get_sig() {
```

Comment:
Could we init the Rand when first call the function of `rand` or `rand_with_seed` like tidb did?

Reply:
ScalarFunc is not passed as mutable reference to function. If we init rand at function first call, I would have to use RefCell two times, like this:

 pub struct ScalarFunc {
    sig: ScalarFuncSig,
    children: Vec<Expression>,
    tp: FieldType,
    cus_rng: RefCell<Option<CusRng>>,
}

struct CusRng {
    rng: RefCell<XorShiftRng>,
}

Seems a little redundant.

Reply:
Done.